src/main/scala/io/iohk/ethereum/vm/Storage.scala

```diff
@@ -0,0 +1,35 @@
+package io.iohk.ethereum.vm
+
+object Storage {
+
+  def fromSeq(words: Seq[DataWord]): Storage = {
+    val map = words.zipWithIndex.map { case (w, i) => DataWord(i) -> w }.toMap
+    new Storage(map)
+  }
+
+  val Empty: Storage = new Storage()
+}
+
+/** Persistent storage of a transaction. May be viewed as a map of 256 bit keys
+  * to 256 bit values.
+  * This is just a representation of storage changes occuring during program execution. The
+  * actual persistence is not this class's resposibility.
+  */
+class Storage private(private val underlying: Map[DataWord, DataWord] = Map()) {
```

Comment:
It's a matter of taste. I didn't have a strong opinion; Radek chose case. Let's keep it as it is if you don't mind.